### ISSUE TITLE
s_client PSK internal_error

### ISSUE THREAD

**Issue by ic0ns** [openssl/openssl#5202]
Openssl 1.1.1-dev s_client sends an internal error instead of a client hello message when started as follows:

```
./openssl s_client -connect 127.0.0.1:4433 -tls1_2 -psk_identity Client_identity -psk 1a2b3c4d

139676759724480:error:142120DB:SSL routines:tls_construct_ctos_early_data:bad psk:ssl/statem/extensions_clnt.c:761:
```
This worked in openssl 1.1.0f.

**Comment by kaduk** [comment]
Did you configure with or without ```enable-tls1_3```?

**Comment by ic0ns** [comment]
The problem seems to appear in both cases.

**Comment by kaduk** [comment]
(I see it with ```enable-tls1_3```, thanks)

**Comment by kaduk** [comment]
For some reason ```TLSEXT_TYPE_early_data``` was not marked as ```SSL_EXT_TLS1_3_ONLY```, which is at least part of the problem here.

**Comment by kaduk** [comment]
I think commit 5ffff5990caa42b8a2d55fc70e23edbe9397e4cb caused us to always try to interpret the passed ```-psk``` bytes as a TLS 1.3 PSK, which seems like a breaking change as well.
(And I think it is coming back to me that we don't mark the early_data extension as TLS1.3-only so that we can consolidate it as the only place for psk session resumption to happen, even if we end up not sending the extension.)  @mattcaswell does that sound right to you?  My current thinking is that we need to rejigger the logic in ```tls_construct_ctos_early_data()``` to not send an alert (and not send the extension) in this case but still attempt to restore a session, and possibly also to make s_client smarter about which callbacks it uses when.

**Comment by mattcaswell** [comment]
Looking at this now. I think the missing `SSL_EXT_TLS1_3_ONLY` just sounds like a mistake to me.

**Comment by mattcaswell** [comment]
Fix in #5205
